veps/sig-storage/dic-on-heterogeneous-cluster/dic-on-heterogeneous-cluster.md

```diff
@@ -0,0 +1,458 @@
+# VEP #NNNN: Golden Images Support For Heterogeneous Clusters
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+- [ ] (R) Target version is explicitly mentioned and approved
+- [ ] (R) Graduation criteria filled
+
+## Overview
+
+Golden images are commonly used OS boot disk images that are used to create virtual machines (VMs) in a Kubernetes
+cluster. Their purpose is to ensure these images automatically available and kept up-to-date. The original design of the
+golden images is documented in the [kubevirt/community repository](https://github.com/kubevirt/community/blob/69d061862e0839608932d225a728a7a6e7a89f29/design-proposals/golden-image-delivery-and-update-pipeline.md).
+
+The initial design assumed homogeneous clusters, where all nodes in the cluster share the same architecture. However, as
+there is a need to support heterogeneous clusters, where nodes may have different architectures (e.g., `arm64`, `amd64`,
+`s390x`), this assumption does not apply anymore, and some changes are required to support this use-case.
+
+This enhancement proposal addresses the challenges of managing golden images in heterogeneous Kubernetes clusters.
+
+## Motivation
+
+The high level flow of the golden images is as follows:
+
+1. The HyperConverged Cluster Operator (HCO) image, includes predefined DataImportCronTemplate files. HCO generates
+   a list of `DataImportCronTemplate` objects in the `SSP` CR, based on these files.
+2. The SSP operator creates `DataImportCron` CRs from the `DataImportCronTemplate` objects in the `SSP` CR.
+3. Either the SSP operator or the CDI operator creates a `DataSource` CRs based on the `DataImportCron` CRs.
```

Comment:
@fossedihelm Would the templates use a platform specific name for the default datasource in a heterogeneous cluster?

Reply:
This is part of this suggestion, yes.

Reply:
Got it, this described in
`For each architecture-specific template, the SSP operator will set the DATA_SOURCE_NAME to the name of the architecture-specific DataSource. The DataSource can be identified using its template.kubevirt.io/architecture and the cdi.kubevirt.io/storage.import.datasource-name labels.`
Is this acceptable for UI?

Reply:
UI will need to support multiple, arch-specific templates anyway. And this specific change is transparent for the UI - the UI does not modify the `DATA_SOURCE_NAME` parameter.

Reply:
@metalice - please keep me honest here.